packages/common/src/visualizations/PieChartDataTransformer.ts

```diff
@@ -47,11 +37,9 @@ export class PieChartDataTransformer<TBarChartLayout, TPieChartConfig> {
                     radius: display?.isDonut ? ['30%', '70%'] : '50%',
                     center: ['50%', '50%'],
                     data: transformedData?.results.map((result) => ({
-                        name: result[
-                            fieldConfig?.groupFieldIds?.[0] || ''
-                        ] as string,
-                        groupId: fieldConfig?.groupFieldIds?.[0] || '',
-                        value: result[fieldConfig?.metricId || ''] as number,
+                        name: result[transformedData.indexColumn.reference],
```

Comment:
note previously there could be many group field ids. Now we use the `index` (e.g. `x` on bar chart) property on `PivotLayout`and there can be only one (for now).

Alternative is to use the `pivot` property instead (e.g. colorBy on bar chart) but I don't think that makes as much sense